anomalib/data/utils/split.py

```diff
@@ -11,76 +11,97 @@
-        seed (int, optional): Random seed to ensure reproducibility. Defaults to 0.
-        normal_label (str): Name of the normal label. For MVTec AD, for instance, this is normal_label.
-
-    Returns:
-        DataFrame: Output dataframe where the part of the training set is assigned to test set.
-    """
-
-    if seed is not None:
-        random.seed(seed)
-
-    normal_train_image_indices = samples.index[(samples.split == "train") & (samples.label == normal_label)].to_list()
-    num_normal_train_images = len(normal_train_image_indices)
-    num_normal_valid_images = int(num_normal_train_images * split_ratio)
+    TRAIN = "train"
+    VAL = "val"
+    TEST = "test"
 
-    indices_to_split_from_train_set = random.sample(population=normal_train_image_indices, k=num_normal_valid_images)
-    samples.loc[indices_to_split_from_train_set, "split"] = "test"
 
-    return samples
+class ValSplitMode(str, Enum):
+    """Splitting mode used to obtain validation subset."""
 
+    SAME_AS_TEST = "same_as_test"
+    FROM_TEST = "from_test"
 
-def create_validation_set_from_test_set(
-    samples: DataFrame, seed: Optional[int] = None, normal_label: str = "good"
-) -> DataFrame:
-    """Craete Validation Set from Test Set.
 
-    This function creates a validation set from test set by splitting both
-    normal and abnormal samples to two.
+def concatenate_datasets(datasets: Sequence[AnomalibDataset]) -> AnomalibDataset:
+    """Concatenate multiple datasets into a single dataset object.
 
     Args:
-        samples (DataFrame): Dataframe containing dataset info such as filenames, splits etc.
-        seed (int, optional): Random seed to ensure reproducibility. Defaults to 0.
-        normal_label (str): Name of the normal label. For MVTec AD, for instance, this is normal_label.
-    """
-
-    if seed is not None:
-        random.seed(seed)
+        datasets (Sequence[AnomalibDataset]): Sequence of at least two datasets.
 
-    # Split normal images.
-    normal_test_image_indices = samples.index[(samples.split == "test") & (samples.label == normal_label)].to_list()
-    num_normal_valid_images = len(normal_test_image_indices) // 2
+    Returns:
+        AnomalibDataset: Dataset that contains the combined samples of all input datasets.
+    """
+    concat_dataset = datasets[0]
+    for dataset in datasets[1:]:
+        concat_dataset += dataset
+    return concat_dataset
 
-    indices_to_sample = random.sample(population=normal_test_image_indices, k=num_normal_valid_images)
-    samples.loc[indices_to_sample, "split"] = "val"
 
-    # Split abnormal images.
-    abnormal_test_image_indices = samples.index[(samples.split == "test") & (samples.label != normal_label)].to_list()
-    num_abnormal_valid_images = len(abnormal_test_image_indices) // 2
+def random_split(
+    dataset: AnomalibDataset, split_ratio: Union[float, Sequence[float]], label_aware: bool = False
+) -> List[AnomalibDataset]:
```

Comment:
would be nice to have a `seed` argument here (`torch.randperm` can be replaced by `random.Random.shuffle` if that is easier)